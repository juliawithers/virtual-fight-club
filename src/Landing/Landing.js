import React, { Component } from 'react'
import CreateAccount from '../CreateAccount/CreateAccount'
import Login from '../Login/Login'
import AboutGame from '../AboutGame/AboutGame'
import context from '../context'

export default class Landing extends Component {
    static contextType=context;
    
    // maybe create am "about" compenent for the descriptions
    render(){
        return(
            <div>
                <AboutGame/>
                <CreateAccount/>
                <Login/>
            </div>
        )
    }
}