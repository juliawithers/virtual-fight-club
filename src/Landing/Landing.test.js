import React from 'react';
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReactDOM from 'react-dom';
import Landing from './Landing'

import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

describe(`Landing component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <Landing/>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    })
    it('renders the UI as expected',() =>{
        const wrapper = shallow(<Landing/>)
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})