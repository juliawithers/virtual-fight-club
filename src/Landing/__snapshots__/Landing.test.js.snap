// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component renders the UI as expected 1`] = `
<div>
  <AboutGame />
  <CreateAccount />
  <Login />
</div>
`;
